File: arrays/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
  public int maxProfit(int[] prices) {
    if (prices.length < 2) {
      return 0;
    }

    int maxProfit = 0;
    int min = prices[0];

    for (int i = 1; i < prices.length; i++) {
      if (prices[i] <= min) {
        min = prices[i];
      } else {
        maxProfit = Math.max(maxProfit, prices[i] - min);
      }
    }
    return maxProfit;
  }
}

File: arrays/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
package besttimetobuyandsellstock

func maxProfit(prices []int) int {
	if len(prices) < 2 {
		return 0
	}

	maxProfit := 0
	min := prices[0]

	for i := 1; i < len(prices); i++ {
		if prices[i] <= min {
			min = prices[i]
		} else {
			if prices[i]-min > maxProfit {
				maxProfit = prices[i] - min
			}
		}
	}
	return maxProfit
}
